src/Symfony/Bridge/PhpUnit/Legacy/SymfonyTestsListenerTrait.php

```diff
@@ -220,15 +222,17 @@ public function startTest($test)
             if (isset($annotations['class']['expectedDeprecation'])) {
                 $test->getTestResultObject()->addError($test, new AssertionFailedError('`@expectedDeprecation` annotations are not allowed at the class level.'), 0);
             }
-            if (isset($annotations['method']['expectedDeprecation'])) {
-                if (!\in_array('legacy', $groups, true)) {
-                    $this->error = new AssertionFailedError('Only tests with the `@group legacy` annotation can have `@expectedDeprecation`.');
+            if (($hasExpectedDeprecationAnnotation = isset($annotations['method']['expectedDeprecation'])) || $this->checkNumAssertions = \in_array(SymfonyExpectDeprecationTrait::class, (new \ReflectionClass($test))->getTraitNames(), true)) {
+                if ($hasExpectedDeprecationAnnotation) {
+                    self::$expectedDeprecations = $annotations['method']['expectedDeprecation'];
+                    self::$previousErrorHandler = set_error_handler([$this, 'handleError']);
```

Comment:
`s/$this/static::class/` for consistency with the other changes?

Reply:
`self::class`? Everything is `self` in the listener trait.